File: check_json.py
import re, json

with open('android/assets/jsons/Civ VI/Terrains.json', 'r', encoding='utf-8') as f:
    content = f.read()

content_no_comments = re.sub(r'//.*?\n', '\n', content)
content_no_comments = re.sub(r'/\*.*?\*/', '', content_no_comments, flags=re.DOTALL)

data = json.loads(content_no_comments)
land_terrains = [t for t in data if t.get('type') == 'Land']
print(f'Total terrains: {len(data)}')
print(f'Land terrains: {len(land_terrains)}')
for t in land_terrains:
    print(f'  - {t["name"]}')

File: check_json.js
const fs = require('fs');
const path = 'D:/lamviec/test/uncivSAU/android/assets/jsons/Civ VI/Civics.json';
let content = fs.readFileSync(path, 'utf8');

let depth = 0;
let inString = false;
let escape = false;
for (let i = 0; i < content.length; i++) {
  const c = content[i];
  if (inString) {
    if (escape) escape = false;
    else if (c === '\\') escape = true;
    else if (c === '"') inString = false;
    continue;
  }
  if (c === '"') inString = true;
  else if (c === '[') depth++;
  else if (c === ']') depth--;
  else if (c === '{') depth++;
  else if (c === '}') depth--;
  
  if (depth < 0) {
    console.log('Unbalanced ] at', i, 'context:', content.substring(i-20, i+20));
    break;
  }
}
console.log('Final depth:', depth);